docs/user-guide.md

```diff
@@ -90,13 +89,41 @@ be too. You might use this method like this:
 Tools could then use the `@Nullable` information to determine that the first use
 is safe but the second is not.
 
+As far as JSpecify is concerned, `String` and `@Nullable String` are _different_
+types. A variable of type `String` can reference any string. A variable
+of type `@Nullable String` can too, but it can also be null. This means that
+`String` is a _subtype_ of `@Nullable String`, in the same way that `Integer` is
+a subtype of `Number`. One way to look at this is that a subtype narrows the
+range of possibilities. A `Number` variable can be assigned from an `Integer`
+but it can also be assigned from a `Long`. Meanwhile an `Integer` variable
+can't be assigned from a `Number` (since that `Number` might be a `Long` or some
+other subtype). Likewise, a `@Nullable String` can be assigned from a
+`String` but a `String` can't be assigned from a `@Nullable String` (since that
+might be null).
+
+```java
+@NullMarked
+class Example {
+  void useNullable(@Nullable String x) {...}
+  void useNonNull(String x) {...}
+  void example(@Nullable String nullable, String nonNull) {
+    useNullable(nonNull); // JSpecify allows this
+    useNonNull(nullable); // JSpecify doesn't allow this
+  }
+}
+```
+
 ## `@NullMarked`
 
 The `@NullMarked` annotation indicates that references can't be null in its scope,
 unless their types are explicitly marked `@Nullable`. If applied to a package then
```

Comment:
This doesn't list `module` and is the only place that calls out `interface` separately from `class`.

Reply:
I added the `module` case. I think it's OK that we say "class or interface" here when we just said "class" when we were introducing `@NullMarked` earlier. That was a quick intro, this is more detailed. I don't think there are any other places where we say "class" where we could say "class or interface".